gaia-fast-list.js

```diff
@@ -558,10 +569,25 @@ Internal.prototype = {
     this.model = model;
     this.sections = this.sectionize(model);
     if (!this.fastList) this.createList();
-    else this.fastList.reloadData();
+    else this.reloadData();
     this.setEmpty(false);
   },
 
+  /**
+   * Reloading the list will completely
+   * re-render the list.
+   *
+   * We must empty the image cache as data
+   * may have changed or item indexed
+   * may not longer map to new model.
+   *
+   * @private
+   */
+  reloadData() {
+    this.emptyImageCache();
```

Comment:
That's a really good idea! Filed issue https://github.com/gaia-components/gaia-fast-list/issues/20